integration-tests/shared/mockserver/server.go

```diff
@@ -58,9 +50,23 @@ func (server *MockServer) Start(t *testing.T) {
 }
 
 func ReturnResponse(t *testing.T, responseString string) http.HandlerFunc {
+	return CheckBodyAndReturnResponse(t, responseString, nil, nil)
+}
+
+func CheckBodyAndReturnResponse(t *testing.T, responseString string, expected *[]string, unexpected *[]string) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
-		_, err := io.ReadAll(req.Body)
+		body, err := io.ReadAll(req.Body)
 		require.NoError(t, err)
+		if expected != nil {
+			for _, expectedString := range *expected {
+				require.Contains(t, string(body), expectedString)
+			}
+		}
+		if unexpected != nil {
+			for _, unexpectedString := range *unexpected {
+				require.NotContains(t, string(body), unexpectedString)
+			}
+		}
```

Comment:
If you don't pass `expected` and `unexpected` as pointers, you can simplify this to

```suggestion
		for _, expectedString := range expected {
			require.Contains(t, string(body), expectedString)
		}
		for _, unexpectedString := range unexpected {
			require.NotContains(t, string(body), unexpectedString)
		}
```